From 83c2cf158b160ed27e0c947bfd3b29bd37efa656 Mon Sep 17 00:00:00 2001
From: Ethan Chen <intervigil@gmail.com>
Date: Wed, 2 May 2018 08:56:17 -0700
Subject: [PATCH 8/9] livedisplay: Move extra inclusions out of header files

* Some of these source files have implicit inclusions because their
  header files transitively include other headers that are necessary.

Change-Id: I698e46b8ad0d0dc8fe3e4688030a9546f4354a24
---
 livedisplay/1.0/default/controller/SDMController.cpp       | 2 ++
 livedisplay/1.0/default/impl/SDM.cpp                       | 7 +++++--
 livedisplay/1.0/default/include/Color.h                    | 7 +++++--
 livedisplay/1.0/default/include/ColorBackend.h             | 4 ++--
 livedisplay/1.0/default/include/Utils.h                    | 1 -
 livedisplay/1.0/default/include/controller/SDMController.h | 4 +++-
 livedisplay/1.0/default/include/impl/SDM.h                 | 1 -
 livedisplay/1.0/default/src/Color.cpp                      | 4 +++-
 livedisplay/1.0/default/src/Utils.cpp                      | 4 ++--
 9 files changed, 22 insertions(+), 12 deletions(-)

diff --git a/livedisplay/1.0/default/controller/SDMController.cpp b/livedisplay/1.0/default/controller/SDMController.cpp
index 0d575ef..1355ce2 100644
--- a/livedisplay/1.0/default/controller/SDMController.cpp
+++ b/livedisplay/1.0/default/controller/SDMController.cpp
@@ -16,6 +16,8 @@
 
 #include "controller/SDMController.h"
 
+#include <android-base/logging.h>
+
 #include <dlfcn.h>
 
 #define LOAD_SDM_FUNCTION(name) \
diff --git a/livedisplay/1.0/default/impl/SDM.cpp b/livedisplay/1.0/default/impl/SDM.cpp
index c14f75e..a7d4abf 100644
--- a/livedisplay/1.0/default/impl/SDM.cpp
+++ b/livedisplay/1.0/default/impl/SDM.cpp
@@ -19,10 +19,13 @@
 
 #define LOG_TAG "LiveDisplay-SDM"
 
-#include <android-base/properties.h>
+#include "impl/SDM.h"
 
 #include "Utils.h"
-#include "impl/SDM.h"
+#include "controller/SDMController.h"
+
+#include <android-base/logging.h>
+#include <android-base/properties.h>
 
 namespace {
 constexpr int kDppsBufSize = 64;
diff --git a/livedisplay/1.0/default/include/Color.h b/livedisplay/1.0/default/include/Color.h
index 57ea6ab..cb53e6d 100644
--- a/livedisplay/1.0/default/include/Color.h
+++ b/livedisplay/1.0/default/include/Color.h
@@ -18,10 +18,11 @@
 #ifndef VENDOR_LINEAGE_LIVEDISPLAY_V1_0_COLOR_H
 #define VENDOR_LINEAGE_LIVEDISPLAY_V1_0_COLOR_H
 
-#include <utils/Mutex.h>
 #include <vendor/lineage/livedisplay/1.0/IColor.h>
 
-#include "ColorBackend.h"
+#include <utils/Mutex.h>
+
+#include <memory>
 
 namespace vendor {
 namespace lineage {
@@ -37,6 +38,8 @@ using ::android::sp;
 
 using ::vendor::lineage::livedisplay::V1_0::IColor;
 
+class ColorBackend;
+
 class Color : public IColor {
   public:
     Color();
diff --git a/livedisplay/1.0/default/include/ColorBackend.h b/livedisplay/1.0/default/include/ColorBackend.h
index b15df58..2d7b571 100644
--- a/livedisplay/1.0/default/include/ColorBackend.h
+++ b/livedisplay/1.0/default/include/ColorBackend.h
@@ -17,10 +17,10 @@
 #ifndef VENDOR_LINEAGE_LIVEDISPLAY_V1_0_COLORBACKEND_H
 #define VENDOR_LINEAGE_LIVEDISPLAY_V1_0_COLORBACKEND_H
 
-#include <vector>
-
 #include "Types.h"
 
+#include <vector>
+
 namespace vendor {
 namespace lineage {
 namespace livedisplay {
diff --git a/livedisplay/1.0/default/include/Utils.h b/livedisplay/1.0/default/include/Utils.h
index 52e3a11..497de7d 100644
--- a/livedisplay/1.0/default/include/Utils.h
+++ b/livedisplay/1.0/default/include/Utils.h
@@ -18,7 +18,6 @@
 #ifndef VENDOR_LINEAGE_LIVEDISPLAY_V1_0_UTILS_H
 #define VENDOR_LINEAGE_LIVEDISPLAY_V1_0_UTILS_H
 
-#include <stdlib.h>
 #include <utils/Errors.h>
 
 using ::android::status_t;
diff --git a/livedisplay/1.0/default/include/controller/SDMController.h b/livedisplay/1.0/default/include/controller/SDMController.h
index 10a4590..2e634de 100644
--- a/livedisplay/1.0/default/include/controller/SDMController.h
+++ b/livedisplay/1.0/default/include/controller/SDMController.h
@@ -17,7 +17,9 @@
 #ifndef VENDOR_LINEAGE_LIVEDISPLAY_V1_0_SDMCONTROLLER_H
 #define VENDOR_LINEAGE_LIVEDISPLAY_V1_0_SDMCONTROLLER_H
 
-#include <android-base/logging.h>
+#include <memory>
+
+#include <stdint.h>
 
 namespace vendor {
 namespace lineage {
diff --git a/livedisplay/1.0/default/include/impl/SDM.h b/livedisplay/1.0/default/include/impl/SDM.h
index b4b8002..4ceeca8 100644
--- a/livedisplay/1.0/default/include/impl/SDM.h
+++ b/livedisplay/1.0/default/include/impl/SDM.h
@@ -19,7 +19,6 @@
 #define VENDOR_LINEAGE_LIVEDISPLAY_V1_0_SDM_H
 
 #include "ColorBackend.h"
-#include "controller/SDMController.h"
 
 #include <vector>
 
diff --git a/livedisplay/1.0/default/src/Color.cpp b/livedisplay/1.0/default/src/Color.cpp
index 77ecae1..9f46f0f 100644
--- a/livedisplay/1.0/default/src/Color.cpp
+++ b/livedisplay/1.0/default/src/Color.cpp
@@ -21,12 +21,14 @@
 #include "Color.h"
 
 #include "ColorBackend.h"
-#if defined(COLOR_BACKEND_SDM)
+#ifdef COLOR_BACKEND_SDM
 #include "impl/SDM.h"
 #elif defined(COLOR_BACKEND_LEGACYMM)
 #include "impl/LegacyMM.h"
 #endif
 
+#include <android-base/logging.h>
+
 namespace {
 
 using vendor::lineage::livedisplay::V1_0::DisplayMode;
diff --git a/livedisplay/1.0/default/src/Utils.cpp b/livedisplay/1.0/default/src/Utils.cpp
index 38dbe71..683b667 100644
--- a/livedisplay/1.0/default/src/Utils.cpp
+++ b/livedisplay/1.0/default/src/Utils.cpp
@@ -15,6 +15,8 @@
 ** limitations under the License.
 */
 
+#include "Utils.h"
+
 #include <fcntl.h>
 #include <poll.h>
 #include <signal.h>
@@ -27,8 +29,6 @@
 
 #include <cutils/sockets.h>
 
-#include "Utils.h"
-
 constexpr char LOCAL_STORAGE_PATH[] = "/data/vendor/display";
 constexpr char LOCAL_MODE_ID[] = "livedisplay_mode";
 constexpr char LOCAL_INITIAL_MODE_ID[] = "livedisplay_initial_mode";
-- 
2.17.0

